microsoft/testsuites/dpdk/dpdknffgo.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List, Type
+
+from lisa.executable import Tool
+from lisa.operating_system import Debian
+from lisa.tools import Echo, Gcc, Git, Make, Tar, Wget
+from lisa.util import UnsupportedDistroException
+
+
+class DpdkNffGo(Tool):
+
+    NFF_GO_SRC_LINK = "https://github.com/intel-go/nff-go.git"
+    REPO_DIR = "nff-go"
+    GO_TAR = "go1.17.6.linux-amd64.tar.gz"
+
+    ubuntu_packages = [
+        "lua5.3-dev",
+        "libpcap-dev",
+        "libelf-dev",
+        "hugepages",
+        "libnuma-dev",
+        "libhyperscan-dev",
+        "liblua5.3-dev",
+        "libmnl-dev",
+        "libibverbs-dev",
+    ]
+
+    @property
+    def command(self) -> str:
+        return "nff-go"
+
+    def _check_exists(self) -> bool:
+        git_path = self.node.working_path.joinpath(self.REPO_DIR)
+        return (
+            self.node.execute(f"test -a {git_path.as_posix()}", shell=True).exit_code
+            == 0
+        )
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Echo, Gcc, Make, Git, Tar, Wget]
+
+    @property
+    def can_install(self) -> bool:
+        # nff-go only implemented for debain in lisav2
+        return isinstance(self.node.os, Debian)
+
+    def _install(self) -> bool:
+        node = self.node
+        git = node.tools[Git]
+        echo = node.tools[Echo]
+        wget = node.tools[Wget]
+        tar = node.tools[Tar]
+        self.make = node.tools[Make]
+        gcc = node.tools[Gcc]
+        gcc.install()
+        self.make.install()
+        # get nff-go source and go binaries
+        self.nff_go_path = git.clone(
+            "https://github.com/intel-go/nff-go.git",
+            cwd=node.working_path,
+            dir_name=self.REPO_DIR,
+        )
+        go_tar_path = wget.get(
+            f"https://go.dev/dl/{self.GO_TAR}",
+            file_path=str(node.working_path),
+            filename=self.GO_TAR,
+        )
+        # unpack and add to path
+        tar.extract(go_tar_path, "/usr/local")
+        node.execute(
+            "go mod download",
+            cwd=self.nff_go_path,
+            update_envs={"PATH": "$PATH:/usr/local/go/bin/"},
```

Comment:
The PATH appears many times, and it's worth to use a variable.

Reply:
Is it changed? It can be something like `self.update_path = f"{self.original_path}:/usr/local/go/bin"`. With this, the original_path is not needed.

Reply:
Should be fixed

Reply:
Is it uploaded?

Reply:
I think yes? This comment window is outdated, if you check the files changed bar it shows the current version

Reply:
This comment asks to have a variable like below, so the below string doesn't need to write three times in code. But it's still had three times.

`self.updated_path = f"{self.original_path}:/usr/local/go/bin/"`